invisible_cities/io/mcinfo_io.py

```diff
@@ -186,26 +250,55 @@ def copy_mc_info(h5in : tb.File, writer : Type[mc_info_writer], which_events : L
     """
     try:
         if which_events is None:
-            which_events = h5in.root.MC.extents.cols.evt_number[:]
-        mcinfo = tbl.get_mc_info(h5in)
-        for n in which_events:
-            writer(mctables=mcinfo, evt_number=n)
+            which_events = get_event_numbers_in_file(file_in)
+        mctbls = get_mc_tbl_list(file_in)
+        for tbl in mctbls:
+            save_mc_table(file_in,       writer,
+                          tbl    , which_events,
+                          db_file,   run_number)
     except tb.exceptions.NoSuchNodeError:
-        raise tb.exceptions.NoSuchNodeError(f"No MC info in file {h5in}.")
+        raise tb.exceptions.NoSuchNodeError(f"No MC info in file {file_in}.")
     except IndexError:
-        raise IndexError(f"No event {n} in file {h5in}.")
+        raise IndexError(f"No events {which_events} in file {file_in}.")
     except NoParticleInfoInFile as err:
         print(f"Warning: {h5in}", err)
         pass
 
 
-def load_mchits(file_name: str,
-                event_range=(0, int(1e9))) -> Mapping[int, Sequence[MCHit]]:
+def get_mc_tbl_list(file_name: str) -> List[MCTableType]:
+    """
+    Returns a list of the tables in
+    the MC group of a given file
+    """
+    with tb.open_file(file_name, 'r') as h5in:
+        mc_group = h5in.root.MC
+        return [MCTableType[tbl.name] for tbl in mc_group]
 
-    with tb.open_file(file_name, mode='r') as h5in:
-        mchits_dict = read_mchit_info(h5in, event_range)
 
-    return mchits_dict
+def is_oldformat_file(file_name : str) -> bool:
+    with tb.open_file(file_name) as h5in:
+        return hasattr(h5in.root, 'MC/extents')
+
+
+def get_event_numbers_in_file(file_name: str) -> np.ndarray:
+    if is_oldformat_file(file_name):
+        with tb.open_file(file_name, 'r') as h5in:
+            evt_list = h5in.root.MC.extents.cols.evt_number[:]
+        return evt_list
+    else:
+        def _try_unique_evt_itr(group, itr):
+            for elem in itr:
+                try:
+                    yield np.unique(getattr(group, elem).cols.event_id)
+                except tb.exceptions.NoSuchNodeError:
+                    pass
+
+        with tb.open_file(file_name, 'r') as h5in:
+            mc_tbls  = ['hits', 'particles', 'sns_response']
+            evt_list = list(_try_unique_evt_itr(h5in.root.MC, mc_tbls))
```

Comment:
do we understand why are there events that are not in particles or hits table?

Reply:
This is a question for the NEXUS authors really.
There are certain types of simulations that have some and not other MC types saved, a simulation of optical photons for example would have an empty hits table as far as I understand.

Reply:
There are different kinds of simulations, where we save different kinds of output (some will save only sensor response information and not true particles, sometimes the opposite happen, for instance). We want to be able to read all kind of MC files.

Reply:
An option would be to resurrect the old `events` table, which contained two columns, namely the eventID and the total deposited true energy, but I don't know if it's worth it...